File: HLC01_03Java/src/main/java/org/example/reservaHotel02/HotelMenu.java
package org.example.reservaHotel02;

public class HotelMenu {
    private final Console console;
    private final Hotel hotel;

    public HotelMenu() {
        this.console = new Console();
        this.hotel = new Hotel(console);
    }

    private String askForOption() {
        console.showMessage("1 para hacer una reserva\n" +
                "2 para cancelar una reserva\n" +
                "3 para ver todas las reservas\n" +
                "0 para salir\n" +
                "---------> ", true);
        String option = console.readString();  // Utilizamos readString en lugar de readln
        console.showMessage("", true);
        return option;
    }

    private void performOption(String option) {
        switch (option) {
            case "0":
                System.out.println("Adios");
                break;
            case "1":
                hotel.doReservation();
                break;
            case "2":
                hotel.cancelReservation();
                break;
            case "3":
                hotel.showBookings(null);
                break;
            default:
                console.showMessage("La opción introducida no existe...", true);
                break;
        }
    }

    public void executeMenu() {
        String option = "hola";
        try {
            while (!option.equals("0")) {
                option = askForOption();
                performOption(option);
            }
        } catch (Exception e) {
            System.out.println("*** ERROR *** - " + e);
        }
    }

}

File: HLC01_03Kotlin/src/main/kotlin/reservaHotel02/HotelMenu.kt
package org.example.reservaHotel02

class HotelMenu(private val console: Console = Console()) {
    private val hotel = Hotel()


    private fun askForOption(): String {
        console.showMessage("1 para hacer una reserva\n" +
                " 2 para cancelar una reserva\n" +
                "3 para ver todas las reservas\n" +
                "0 para salir\n" +
                "---------> ", true)
        val option = readln()
        console.showMessage("", true)
        return option
    }


    private fun performOption(option: String) {
        when (option) {
            "0" -> println("Adios")
            "1" -> hotel.doReservation()
            "2" -> hotel.cancelReservation()
            "3" -> hotel.showBookings()
            else -> console.showMessage("La opción introducida no existe...")
        }
    }


    fun executeMenu() {
        var option = "hola"
        try {
            while (option != "0") {
                option = askForOption()
                performOption(option)
            }

        } catch (e: Exception) {
            println("*** ERROR *** - $e")
        }
    }

}